src/lava/magma/compiler/builder.py

```diff
@@ -354,14 +354,19 @@ def build(self):
         """
 
         # Create the ProcessModel
-        pm = self.proc_model()
-        pm.model_id = self._model_id
-
         # Default value of pm.proc_params in ProcessModel is an empty dictionary
         # If a proc_params argument is provided in PyProcessBuilder,
         # this will be carried to ProcessModel
-        if self.proc_params is not None:
-            pm.proc_params = self.proc_params
+        if self.proc_params:
```

Comment:
This is a bit ugly. 
AbstractProcModel should just accept proc_params.
Then this conditional is not needed.

The AbstractProcess can just initialize proc_params in its constructor with an empty dictionary.
This ensures that this parameter always exists. Some sub classes of AbstractProc may then chose to actually set it.

Reply:
Resolved